client/my-sites/activity/activity-log/index.jsx

```diff
@@ -147,6 +171,41 @@ class ActivityLog extends Component {
 		}
 	};
 
+	onScroll = () => {
+		const y = window.scrollY;
+
+		if ( ! this.state.scrollTicking ) {
+			// It's best practice to throttle scroll event for performance
+			window.requestAnimationFrame( () => {
+				this.stickFilterBar( y );
+				this.setState( { scrollTicking: false } );
+			} );
+
+			this.setState( { scrollTicking: true } );
+		}
+	};
+
+	stickFilterBar = ( scrollY ) => {
+		const { initialFilterBarY, masterBarHeight } = this.state;
+		const filterBar = this.filterBarRef.current;
+
+		if ( ! filterBar ) {
+			return;
+		}
+
+		if ( ! initialFilterBarY ) {
+			this.setState( { initialFilterBarY: filterBar.getBoundingClientRect().top } );
```

Comment:
This seems like the safest place to get the initial y position of the filter bar, since banners can be rendered after this component has mounted, and therefore move the filter bar down. It's not ideal though, since the same issue can occur if the user scrolls right away.

A further improvement could be to look at the Intersection Observer API.